### ISSUE TITLE
Structured contact details for shows/socs/venues?

### ISSUE THREAD

**Issue by philosophicles** [camdram/camdram#287]
Camdram V1 has a standard bit of text at the bottom of society pages like:

    If you have any queries about this society please contact <John Smith>

Where the person's name was a mailto link to the person's email.  If no people were available to list the line was omitted.  The same was available for venues but not really used as much.

Similarly, shows in V1 also had a standard bit of text:

    This entry is administered by <John Smith>

Which defaulted to the society name if no people were available to list, as a link to the society's page.

Each of these behaviours is controlled by a bit flag against administrators of the society/show object - acts_access.contact in the database.  From the V1 front-end, there's an 'Add to contacts' option on the access control page for an object.

There are dubious aspects to the design of this in V1, but do we want to recreate / replicate the spirit of this in V2 at all? Currently, lots of societies/venues don't have any admins let alone any marked as being 'contact people' - but if done right, there'd be an incentive for that to change, perhaps.  (V1 made email addresses directly visible, which we might want to avoid.)

The alternative is that the society/venue/show text just says who to get in touch with, manually - which is perfectly possible.

**Comment by hoyes** [comment]
Maybe nicest would be to have a "Contact this show" link or something, which takes you to a contact form. Then Camdram could proxy the message to the relevant person/people:
 * Show owner(s)
 * If no show owners, society and venue owners if applicable
 * If no show owners, society or venue (for some inexplicable reason), Camdram admins.

**Comment by philosophicles** [comment]
Sounds good to me; "Contact this society" and "Contact this venue" links would exist similarly with similar but simpler logic behind them.

Rather than "If no show owners, society or venue (for some inexplicable reason).." it's probably "If no show owners, society owners or venue owners..." -- if there's no show owner, there's a good chance there won't be a soc or venue owner as we have plenty of socs/venues without any admins.

**Comment by hoyes** [comment]
Ah yes good point.

**Comment by hoyes** [comment]
I've implemented this and pushed it to Tiberius.
 * All shows newer than 1 year ago have the "Contact" link, whether or not they have any admins. If the show doesn't have any admins, it goes to the society admins. If there's no society or the society doesn't have any admins, then it goes to the venue admins. If there's no venue or venue admins then it goes to websupport. I added the expiry of the 'contact us' link to stop people trying to e-mail shows from many years ago where the admin's e-mail addresses are very likely no longer valid.
 * Venues and societies only have the "Contact ..." link if
 * I've changed all the contact forms (including the main "Contact Us" form) to only prompt for the name and e-mail address if the user is logged out - if logged in these are automatically populated from the user. Also, the captcha isn't used if you're logged in.

Bounces will end up at websupport@camdram.net for now as for support e-mails.

Comments would be appreciated before this goes live, e.g.:
 * Wording and location of the contact links.
 * Implications of people being able to contact societies and venues via Camdram. I hid the links for organisations that don't have admins, as I don't think webteam want to be in the business of forwarding e-mails to the Arts Theatre or West Road. For those with admins, I'm wondering whether people might want some control over which admins receive the e-mails (rather than it going to all admins), but it's probably OK as it is now for iteration one.

**Comment by philosophicles** [comment]
Can't see it for socs/venues on Tiberius because none of them have admins - but it looks fine for shows so I'd say wording/location AOK. 

Your logic re "shows newer than 1 year" also makes good sense, plus I think the general contact form changes are very sensible. Together this stuff should make a big difference, even though it's a only a few words here and there!

In terms of your last bullet point @hoyes, Camdram V1 allowed admins to choose which admins were contactable, but the whole V1 approach exposed individual email addresses (obviously not ideal). Since this uses a form and email addresses aren't harvestable, I'd say it's OK for all admins to get the email - at least until somebody asks us if they can stop getting the emails...?

**Comment by ajf58** [comment]
The logic all sounds good to me. My only comment/question is that whilst this was a feature of V1, V2 has been out for a while, so should we make sure admins/contacts are aware about this element of being an admin? I imagine it's ok (the email address isn't visible on the client side) but it'd be nice to keep everyone informed.

**Comment by hoyes** [comment]
I agree it would be nice to inform people, but I'm not sure how...send an e-mail to all admins?

**Comment by hoyes** [comment]
I think I'm going to do a release tonight, with this included, as in balance there are a few fixes bundled with this that ought to be pushed out.

I've just sent an e-mail to webteam@, which is in no way related to your point @ajf58 ...

**Comment by philosophicles** [comment]
Incidentally, might also be worth pointing out that V1 admin interface does/did include shortcuts for bulk-emailing all current admins.. (very bottom option, oft-neglected and probably not worth including in V2?).

**Comment by philosophicles** [comment]
This has been live for a while now and this issue should probably be closed already - however I'm going to take advantage of it being open to raise a suggestion for improvement. 

There have been a few instances now of people using this to contact show organisers, and already the email addresses of some/all show organisers being extinct. (With cam email addresses, of course, they don't become invalid gradually over time, but in a big lump each summer).

To increase the likelihood of somebody actually responding to the author of the query - would it be better for *all* show enquiries (whenever the 'Contact the organisers' link is displayed) to go to both show and society admins? (Rather than just failing back to soc admins if no show admins exist)? 

My rationale for bothering with this is that, if somebody uses the link to get in touch with a show, and nobody ever responds, it makes Camdram as a site look bad. 

An alternative (more complex?) approach might be to somehow trap email bounces as a result of such enquiries, and either pass them on to the enquirer (ugly messages) or formulate an auto-reply email to the enquirer saying roughly "Camdram tried to get in touch with your query below but some or all of our contacts appear to be out of date. Sorry, there's not much we can do about this."
